client/rest-high-level/src/main/java/org/elasticsearch/client/ml/PostDataRequest.java

```diff
@@ -0,0 +1,224 @@
+    public static final ParseField RESET_END = new ParseField("reset_end");
+    public static final ParseField CONTENT_TYPE = new ParseField("content_type");
+
+    public static final ConstructingObjectParser<PostDataRequest, Void> PARSER =
+        new ConstructingObjectParser<>("post_data_request",
+            (a) -> new PostDataRequest((String)a[0], XContentType.fromMediaTypeOrFormat((String)a[1])));
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), Job.ID);
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), CONTENT_TYPE);
+        PARSER.declareString(PostDataRequest::setResetEnd, RESET_END);
+        PARSER.declareString(PostDataRequest::setResetStart, RESET_START);
+    }
+
+    private final String jobId;
+    private final XContentType xContentType;
+    private final List<BytesReference> bytesReferences;
+    private final List<Map<String, Object>> objectMaps;
+    private String resetStart;
+    private String resetEnd;
+    private BytesReference content;
+
+    /**
+     * PostDataRequest for sending data in JSON format
+     * @param jobId non-null jobId of the job to post data to
+     */
+    public static PostDataRequest postJsonDataRequest(String jobId) {
+        return new PostDataRequest(jobId, XContentType.JSON);
+    }
+
+    /**
+     * PostDataRequest for sending data in SMILE format
+     * @param jobId non-null jobId of the job to post data to
+     */
+    public static PostDataRequest postSmileDataRequest(String jobId) {
+        return new PostDataRequest(jobId, XContentType.SMILE);
+    }
+
+    /**
+     * Create a new PostDataRequest object
+     *
+     * @param jobId non-null jobId of the job to post data to
+     * @param xContentType content type of the data to post. Only {@link XContentType#JSON} {@link XContentType#SMILE} are supported
+     */
+    public PostDataRequest(String jobId, XContentType xContentType) {
+        this.jobId = Objects.requireNonNull(jobId, "job_id must not be null");
+        this.xContentType = Objects.requireNonNull(xContentType, "content_type must not be null");
+        this.bytesReferences = new ArrayList<>();
+        this.objectMaps = new ArrayList<>();
+    }
+
+    public String getJobId() {
+        return jobId;
+    }
+
+    public String getResetStart() {
+        return resetStart;
+    }
+
+    /**
+     * Specifies the start of the bucket resetting range
+     *
+     * @param resetStart String representation of a timestamp; may be an epoch seconds, epoch millis or an ISO string
+     */
+    public void setResetStart(String resetStart) {
+        this.resetStart = resetStart;
+    }
+
+    public String getResetEnd() {
+        return resetEnd;
+    }
+
+    /**
+     * Specifies the end of the bucket resetting range
+     *
+     * @param resetEnd String representation of a timestamp; may be an epoch seconds, epoch millis or an ISO string
```

Comment:
Might be best to say ISO 8601 instead of just ISO.

Also, this can be set to the string "now".